src/System.Diagnostics.Process/tests/ProcessTests.netcoreapp.cs

```diff
@@ -119,5 +125,182 @@ public void BothArgumentSetAndArgumentListSet()
             testProcess.StartInfo = psi;
             Assert.Throws<InvalidOperationException>(() => testProcess.Start());
         }
+
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.Uap)]
+        public void VerifyingCiBehavior()
+        {
+            throw new Exception();
+        }
+
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.Uap)]
+        public void Kill_EntireProcessTree_ThrowsPlatformNotSupportedException()
+        {
+            var process = new Process();
+            Assert.Throws<PlatformNotSupportedException>(() => process.Kill(entireProcessTree: true));
+            Assert.Throws<PlatformNotSupportedException>(() => process.Kill(entireProcessTree: false));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Uap)]
+        public void Kill_EntireProcessTree_True_ProcessNotStarted_ThrowsInvalidOperationException()
+        {
+            var process = new Process();
+            Assert.Throws<InvalidOperationException>(() => process.Kill(entireProcessTree: true));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Uap)]
+        public async Task Kill_EntireProcessTree_False_OnlyRootProcessTerminated()
+        {
+            IReadOnlyList<Process> tree = null;
+            try
+            {
+                tree = CreateProcessTree();
+                Process parentProcess = tree.First();
+
+                parentProcess.Kill(entireProcessTree: false);
+
+                // Since Kill() is fire-and-forget, wait a moment for it to take effect
+                await Task.Delay(10);
```

Comment:
Sadly relying on a sleep especially a small one is almost guaranteed to lead to sporadic test failures because some CI machines are slow/overloaded and we run so many tests every day so many times.

The pattern we prefer is to test, if failure then sleep and retry, and repeat retries until a certain time has elapsed that's larger than any reasonable period that it could possibly take eg 30 seconds is common. Probably you don't want to try 100 times a second so we'd typically have some crude backoff.

Reply:
Changed to using retry w/backoff logic. Right now, retrying ends at just over 10 seconds, which hopefully is enough considering what's being checked. Can increase that, if desired.